src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombinationUtil.java

```diff
@@ -77,4 +88,58 @@ public void validateParams(final Map<String, Object> actualParams, final Set<Str
     public float getWeightForSubQuery(final List<Float> weights, final int indexOfSubQuery) {
         return indexOfSubQuery < weights.size() ? weights.get(indexOfSubQuery) : 1.0f;
     }
+
+    /**
+     * Check if number of weights matches number of queries. This does not apply for case when
+     * weights were not provided, as this is valid default value
+     * @param scores collection of scores from all sub-queries of a single hybrid search query
+     * @param weights score combination weights that are defined as part of search result processor
+     */
+    protected void validateIfWeightsMatchScores(final float[] scores, final List<Float> weights) {
+        if (weights.isEmpty()) {
+            return;
+        }
+        if (scores.length != weights.size()) {
+            log.error(
+                String.format(
+                    Locale.ROOT,
+                    "number of weights [%d] must match number of sub-queries [%d] in hybrid query",
+                    weights.size(),
+                    scores.length
+                )
+            );
+            throw new IllegalArgumentException("number of weights must match number of sub-queries in hybrid query");
+        }
+    }
+
+    /**
+     * Check if provided weights are valid for combination. Following conditions are checked:
+     * - every weight is between 0.0 and 1.0
+     * - sum of all weights must be equal 1.0
+     * @param weightsList
+     */
+    private void validateWeights(final List<Float> weightsList) {
+        boolean isOutOfRange = weightsList.stream().anyMatch(weight -> !Range.between(0.0f, 1.0f).contains(weight));
+        if (isOutOfRange) {
+            log.error(
```

Comment:
Same here. Returning submitted weights in exception should be fine rather than logging duplicated message because we already knows the value is float. And float won't cause any security risk by getting returned in exception.

Reply:
ack